post-merge-review.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+set -e
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+REGEX_REF='^\(cherry picked from commit ([a-f0-9]{40})\)$'
+
+do_reference_check() {
+  if ! [[ $(git log --format=%B "$line^!" | grep -E "$REGEX_REF") =~ $REGEX_REF ]]; then
+    echo -e "${RED}FAIL${NC}" >> "$GITHUB_OUTPUT"
+    dirty=1
+    echo "  Cannot find reference in commit message" >> "$GITHUB_OUTPUT"
+    echo "  Hint: use -x with git cherry-pick" >> "$GITHUB_OUTPUT"
+    return;
+  fi
+
+  if ! eval "$(git branch --contains "${BASH_REMATCH[1]}" "$check_branch" 2>/dev/null)"; then
+    echo -e "${RED}FAIL${NC}" >> "$GITHUB_OUTPUT"
+    dirty=1
+    echo "  Cannot find ${BASH_REMATCH[1]} from $line in $check_branch" >> "$GITHUB_OUTPUT"
+    return;
+  fi
+  echo -e "${GREEN}OK${NC}" >> "$GITHUB_OUTPUT"
+}
+
+do_subject_check() {
+  summary=$(git log --format=%s "$line^!")
+  if (($(git log --format=%s "$check_branch" | grep -Fxc "$summary") == 0)); then
```

Comment:
I really don't understand what's happening here sorry, can you explain?

Reply:
I meant I don't understand where the string "summary" is supposed to come from and why.

Reply:
Oh whoops, got my functions confused.

This check is simply taking the first line of the commit and seeing if that summary exists int he history of main

Reply:
> Oh whoops, got my functions confused.

Then how about renaming this one to something like `is_present_in_main()`?

OK, this is becoming clearer now. These changes would have really helped:

- `line` => `commit=$1` (DONE)
- `summary` => `subject`
-  `"$line^!"` => `-n 1 "$line"`
- `check_branch` => `main_branch` as opposed to the target branch which is always the one "checked" by Github checks.

Thanks for teaching me about `sha^!`, I will for sure re-use it. However I've been the "local git expert" a number of times and I've pointed people at `man git revisions` many times yet I didn't know it. So I suspect many people don't either.